Materials/Materials/Program.cs

```diff
@@ -13,135 +13,38 @@
-		var env = context.RequestServices.GetService<IWebHostEnvironment>();
-		var errorCode = StatusCodes.Status500InternalServerError;
-		context.Response.StatusCode = errorCode;
-		context.Response.ContentType = "application/json";
-
-		var errorMessage = "An unexpected error occurred. Please try again later.";
-		
-		if (exceptionHandlerPathFeature?.Error is ArgumentException)
-		{
-			errorMessage = exceptionHandlerPathFeature.Error.Message;
-			errorCode = StatusCodes.Status400BadRequest;
-		}
-		else if (exceptionHandlerPathFeature?.Error is NotFoundException)
-		{
-			errorMessage = exceptionHandlerPathFeature.Error.Message;
-			errorCode = StatusCodes.Status404NotFound;
-		}
-		else if (exceptionHandlerPathFeature?.Error is ConflictException)
-		{
-			errorMessage = exceptionHandlerPathFeature.Error.Message;
-			errorCode = StatusCodes.Status409Conflict;
-		}
-
-		if (env.IsDevelopment())
-		{
-			// In development, return detailed error information
-			errorMessage += " dev info: " + exceptionHandlerPathFeature?.Error.InnerException;
-		}
-		await context.Response.WriteAsJsonAsync(new 
-		{ 
-			ErrorMessage = errorMessage, 
-			ErrorCode = errorCode 
-		});
-	});
-});
-
-
 // StatusCodePages Middleware
 app.UseStatusCodePages(context =>
 {
 	context.HttpContext.Response.ContentType = "application/json";
 	return context.HttpContext.Response.WriteAsJsonAsync(new { ErrorMessage = "An unexpected error occurred." });
 });
 
+app.UseRouting(); // This must come before UseAuthentication and UseAuthorization
+app.UseHttpsRedirection();
 
-// Routes
-app.MapGet("/", BaseUrl);
-app.MapGet("/material/list", ListOfCourseTemplates);
-app.MapGet("/material/id", GetMaterialById);
-app.MapGet("/material/course", GetMaterialSByCourse);
-app.MapPost("/material/add", AddMaterial);
-
-app.Run();
-
-// Route implementations
-string BaseUrl()
-{
-	return "Hello World!";
-}
-
-async Task<object> ListOfCourseTemplates([FromServices] MaterialService service)
-{
-	var result = await service.ListAsync();
-	return new { Data = result };
-}
-
-static async Task<object> AddMaterial([FromServices] MaterialService service, MaterialDataDtoAdd material)
-{
-	if (string.IsNullOrEmpty(material.Name))
-	{
-		throw new ArgumentException("Invalid data provided. Material name is required.");
-	}
-	if (string.IsNullOrEmpty(material.CourseId))
-	{
-		throw new ArgumentException("Invalid data provided. CourseId is required.");
-	}
-
-	try
-	{ 
-		Convert.ToInt32(material.CourseId);
-	}
-	catch (Exception e)
-	{
-		throw new ArgumentException("Invalid data provided: CourseId.");
-	}
+app.UseMiddleware<CustomErrorHandlingMiddleware>();
 
-	await service.AddAsync(material);
-	return new { Message = "Material added successfully." };
-}
+app.MapGet("/", EndpointHandlers.BaseUrl);
+app.MapGet("/material/list", EndpointHandlers.ListOfCourseTemplates);
+app.MapGet("/material/id", EndpointHandlers.GetMaterialByMaterialId);
```

Comment:
Let's rename this to `/materials/{id}`. It looks like a bit more `REST-ful` approach